src/ol/source/BingMaps.js

```diff
@@ -261,7 +270,17 @@ class BingMaps extends TileImage {
             if (hidpi) {
               url += '&dpi=d1&device=mobile';
             }
-            return url.replace('{quadkey}', quadKey(quadKeyTileCoord));
+            url = url.replace('{quadkey}', quadKey(quadKeyTileCoord));
+            const uri = new URL(url);
+            const params = uri.searchParams;
```

Comment:
Even though the implementation works, I think it is a bit awkward: we start by manipulating the URL as a string, then switch to the `URL` class. I would suggest working with a `URL` object from the start (except for the `{quadkey}` token replacement), and return `url.toString()` in the end. Would that make sens to you?